src/Constant.cpp

```diff
@@ -96,12 +96,55 @@ Constant::clone() const
 	return new Constant(*this);
 }
 
+//--------------------------------------------------------------
+string HexToBinary(string val)
+{
+	std::ostringstream oss;
+	for (unsigned int i = 0; i < val.size(); i++) {
+		if (val[i] == '0')
+			oss << "0000";
+		else if (val[i] == '1')
+			oss<<"0001";
+		else if (val[i] == '2')
+			oss<<"0010";
+		else if (val[i] == '3')
+			oss<<"0011";
+		else if (val[i] == '4')
+			oss<<"0100";
+		else if (val[i] == '5')
+			oss<<"0101";
+		else if (val[i] == '6')
+			oss<<"0110";
+		else if (val[i] == '7')
+			oss<<"0111";
+		else if (val[i] == '8')
+			oss<<"1000";
+		else if (val[i] == '9')
+			oss<<"1001";
+		else if (val[i] == 'A')
+			oss<<"1010";
+		else if (val[i] == 'B')
+			oss<<"1011";
+		else if (val[i] == 'C')
+			oss<<"1100";
+		else if (val[i] == 'D')
+			oss<<"1101";
+		else if (val[i] == 'E')
+			oss<<"1110";
+		else if (val[i] == 'F')
+			oss<<"1111";
+	}
+	return oss.str();
+}
+
 // --------------------------------------------------------------
 static string
 GenerateRandomCharConstant(void)
 {
 	string ch;
-	if (CGOptions::ccomp() || !CGOptions::longlong())
+	if (CGOptions::binary_constant() && rnd_flipcoin(30)) {
```

Comment:
It's preferred to add the probability in Probabilities.h thus enable users to customize it. 30% is probably too high. What do we gain by generating constants represented by `0b` instead of `0x`?

Reply:
Done. I have added probability for binary constant as `3%` and necessary changes.

I am following GCC docs for adding the extensions - `https://gcc.gnu.org/onlinedocs/gcc/Binary-constants.html#Binary-constants`. As per docs, it is just convention to start binary constant with `0b` or `0B'. I don't see any gain in it. 

Would it be good idea if we start binary constants with '0x' in terms of finding bugs?

Reply:
The gain with using 0b instead of 0x, with regard to compiler testing, is probably minimal. Most likely it exercises the parser only.